src/ray/gcs/tables.h

```diff
@@ -61,91 +63,62 @@ class Table {
   ///        GCS.
   /// \return Status
   Status Add(const JobID &job_id, const ID &id, std::shared_ptr<DataT> data,
-             const Callback &done) {
-    auto d = std::shared_ptr<CallbackData>(
-        new CallbackData({id, data, done, nullptr, nullptr, this, client_}));
-    int64_t callback_index =
-        RedisCallbackManager::instance().add([d](const std::string &data) {
-          if (d->callback != nullptr) {
-            (d->callback)(d->client, d->id, d->data);
-          }
-        });
-    flatbuffers::FlatBufferBuilder fbb;
-    fbb.ForceDefaults(true);
-    fbb.Finish(Data::Pack(fbb, data.get()));
-    RAY_RETURN_NOT_OK(context_->RunAsync("RAY.TABLE_ADD", id, fbb.GetBufferPointer(),
-                                         fbb.GetSize(), prefix_, pubsub_channel_,
-                                         callback_index));
-    return Status::OK();
-  }
+             const Callback &done);
 
   /// Lookup an entry asynchronously.
   ///
   /// \param job_id The ID of the job (= driver).
   /// \param id The ID of the data that is looked up in the GCS.
-  /// \param lookup Callback that is called after lookup.
+  /// \param lookup Callback that is called after lookup. If the callback is
+  ///        called with an empty vector, then there was no data at the key.
   /// \return Status
-  Status Lookup(const JobID &job_id, const ID &id, const Callback &lookup,
-                const FailureCallback &failure) {
-    auto d = std::shared_ptr<CallbackData>(
-        new CallbackData({id, nullptr, lookup, failure, nullptr, this, client_}));
-    int64_t callback_index =
-        RedisCallbackManager::instance().add([d](const std::string &data) {
-          if (data.empty()) {
-            if (d->failure != nullptr) {
-              (d->failure)(d->client, d->id);
-            }
-          } else {
-            auto result = std::make_shared<DataT>();
-            auto root = flatbuffers::GetRoot<Data>(data.data());
-            root->UnPackTo(result.get());
-            if (d->callback != nullptr) {
-              (d->callback)(d->client, d->id, result);
-            }
-          }
-        });
-    std::vector<uint8_t> nil;
-    RAY_RETURN_NOT_OK(context_->RunAsync("RAY.TABLE_LOOKUP", id, nil.data(), nil.size(),
-                                         prefix_, pubsub_channel_, callback_index));
-    return Status::OK();
-  }
+  Status Lookup(const JobID &job_id, const ID &id, const Callback &lookup);
```

Comment:
This change breaks `ExecuteGetLocations` in `src/ray/object_manager/object_directory.cc`. Can we update `ExecuteGetLocations` to reflect these changes? The change should be relatively simple. We'll have test coverage for the new object manager soon.